### ISSUE TITLE
High CPU use

### ISSUE THREAD

**Issue by isjamesalive** [Microsoft/vscode#37638]
I've been using Code happily for a few weeks and it's suddenly started chewing up 99% CPU.  The same thing happens for stable builds and the Insider builds, and after deleting my %appdata%/code and ~/.vscode directories.

I've attached startup profiles and a minidump - let me know if I can do anything else.

[8586_main.cpuprofile.txt](https://github.com/Microsoft/vscode/files/1443048/8586_main.cpuprofile.txt)
[8586_renderer.cpuprofile.txt](https://github.com/Microsoft/vscode/files/1443049/8586_renderer.cpuprofile.txt)

[Code.dmp.zip](https://github.com/Microsoft/vscode/files/1443050/Code.dmp.zip)

- VSCode Version: <code>Code 1.17.2 (b813d12980308015bcd2b3a2f6efa5c810c33ba5, 2017-10-16T13:59:46.104Z)</code>
- OS Version: <code>Windows_NT x64 6.1.7601</code>
- CPUs: <code>Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz (1 x 2592)</code>
- Memory (System): <code>4.00GB (2.79GB free)</code>
- Memory (Process): <code>201.53MB working set (202.53MB peak, 179.17MB private, 49.46MB shared)</code>
- Load (avg): <code>0, 0, 0</code>
- VM: <code>100%</code>
- Initial Startup: <code>yes</code>
- Screen Reader: <code>no</code>
- Empty Workspace: <code>yes</code>
- Timings:

|Component|Task|Time (ms)|
|---|---|---|
|main|start => app.isReady|312|
|main|app.isReady => window.loadUrl()|68099|
|renderer|window.loadUrl() => begin to require(workbench.main.js)|35812|
|renderer|require(workbench.main.js)|8862|
|renderer|create extension host => extensions onReady()|7657|
|renderer|restore viewlet|0|
|renderer|restore editor view state|409|
|renderer|overall workbench load|1637|
|main + renderer|start => extensions ready|121762|
|main + renderer|start => workbench ready|114862|

---

:warning: Make sure to **attach** these files from your *home*-directory: :warning:
-`8586_main.cpuprofile.txt`
-`8586_renderer.cpuprofile.txt`

**Comment by bpasero** [comment]
@isjamesalive can you give us more information by going through https://github.com/Microsoft/vscode/wiki/Performance-Issues ?

**Comment by jrieken** [comment]
Funky.... Profiler shows 1 minute idle....

<img width="1119" alt="screen shot 2017-11-06 at 09 40 11" src="https://user-images.githubusercontent.com/1794099/32432106-8479087c-c2d6-11e7-8414-3a1b6d08facc.png">

**Comment by isjamesalive** [comment]
@bpasero Sure. 
> What is the size of your workspace/project (how many files)?
Only 2 files with a combined length of 1000 lines, 36Kb. Including data files, it's 12 files totally 428Kb, but I don't open the data files in the application.
> What languages are used in this project?
PowerShell 5.1
> Do you run Code on a single-CPU environment (e.g. VM)? We know that processes can go crazy and startup times can be super slow.
Yes - it's a VM with 1 core and 4GB RAM provisioned. I'll add more cores and see if that helps if you like.
> Which Code process consumes the most resources?
It's the original parent process - see screenshot.

![qfpabdf](https://user-images.githubusercontent.com/4370270/32435892-3f2e6eb8-c332-11e7-940e-3876e249feec.jpg)

This happens on a clean install and with the official PowerShell extension installed.

**Comment by isjamesalive** [comment]
I added 2 cores and 2GB RAM to this machine last night, and I don't seem to be encountering the problem anymore. I rebooted when prompted after Windows reconfigured itself for the extra resources and I can't reproduce the problem now.

> The process rundll32.exe has initiated the restart of computer WIN7-PC on behalf of user Win7-PC\Win7 for the following reason: Hardware: Installation (Planned)
>  Reason Code: 0x80010002
>  Shutdown Type: restart
>  Comment: 

I checked the event log and no new Windows updates were installed during reboots. Attaching another set of profile logs. 

[e0d4_renderer.cpuprofile.txt](https://github.com/Microsoft/vscode/files/1447982/e0d4_renderer.cpuprofile.txt)
[e0d4_main.cpuprofile.txt](https://github.com/Microsoft/vscode/files/1447983/e0d4_main.cpuprofile.txt)
